misp_stix_converter/stix-to-misp.py

```diff
@@ -1,69 +1,94 @@
+            config = pyaml.yaml.load(f)
+    except FileNotFoundError:
+        log.fatal("Could not find config file %s", cfg_path)
+        return(1)
+
+    # Backwards compatibility, if users haven't updated config
+    if SSL_CONFIG_KEY not in config[MISP_CONFIG_KEY]:
+        log.warning("Please update your config file using the"
+                    " misp.login.example to include " + SSL_CONFIG_KEY)
+        time.sleep(BLAST_FROM_PAST_PENALTY_SECS)
+        config[MISP_CONFIG_KEY][SSL_CONFIG_KEY] = False
+
+    # This is just a file conversion - relatively quick and easy
+    in_path = options.file
+    log.info("Opening STIX file %s", in_path)
+    try:
+        with open(in_path, "r") as f:  # Read binary, as maybe just bytes
+            pkg = load_stix(f)
+    except OSError:
+        log.fatal("Could not open STIX file %s", in_path)
+        return (1)
+
+    misp_args = (
+        config[MISP_CONFIG_KEY]["URL"],
+        config[MISP_CONFIG_KEY]["KEY"],
+        config[MISP_CONFIG_KEY].get(SSL_CONFIG_KEY, True)
+    )
+    planet_misp = misp.MISP(*misp_args)
+
+    log.info("Pushing to MISP...")
+    planet_misp.push(pkg)
+    log.info("COMPLETE")
+
 
-parser = argparse.ArgumentParser(description='Process some integers.')
-parser.add_argument("-c", "--config", help="Path to config file. Default is misp.login.")
-parser.add_argument("-v", "--verbose", help="Increase logging verbosity.", default=False, action="store_true")
-parser.add_argument("file", help="The STIX file to push")
-
-args = parser.parse_args()
-
-# Set up a logger
-log = logging.getLogger(__name__)
-log.setLevel(logging.DEBUG if args.verbose else logging.INFO)
-ch = logging.StreamHandler(sys.stdout)
-ch.setLevel(logging.DEBUG if args.verbose else logging.INFO)
-formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
-ch.setFormatter(formatter)
-log.addHandler(ch)
-
-# Set the config file
-if args.config:
-    configfile = args.config
-else:
-    configfile = os.path.expanduser("~/.misptostix/misp.login")
-
-try:
-    with open(configfile, "r") as f:
-        CONFIG = pyaml.yaml.load(f)
-except FileNotFoundError:
-    log.fatal("Could not find config file %s", configfile)
-    sys.exit(1)
-
-# Backwards compatability, if users haven't updated config
-if "SSL" not in CONFIG["MISP"]:
-    log.warning("Please update your config file using the misp.login.example to include SSL")
-    time.sleep(1)
-    CONFIG["MISP"]["SSL"] = False
-
-# This is just a file conversion
-# Relatively quick and easy
-MISP = misp.MISP(CONFIG["MISP"]["URL"], CONFIG["MISP"]["KEY"], CONFIG["MISP"].get("SSL", True))
-
-# Load the package
-log.info("Opening STIX file %s", args.file)
-# Sometimes it's thrown as bytes?
-fname = args.file
-with open(fname, "r") as f:
-    pkg = load_stix(f)
-
-# We'll use my nice little misp module
-log.info("Pushing to MISP...")
-MISP.push(pkg)
-log.info("COMPLETE")
+if __name__ == '__main__':
+    sys.exit(main)
```

Comment:
[This will just exit with a faliure code](https://docs.python.org/3/library/sys.html#sys.exit)